### PR TITLE
docs: accessibility improvements

### LINKED ISSUE #881: docs: accessibility improvements
### Is your feature request related to a problem? Please describe.

Ben Myers streamed on twitch recently and had a [fantastic section](https://www.twitch.tv/videos/1627977114)(vod link) going over the docs and accessibility concerns with them. I think this would be a good first issue as well, with most being fairly straight forward updates.

### Describe the solution you'd like to see

These are the concerns raised, with the timestamp from the vod above. Ben goes into really great detail on why each thing is a concern, either to keyboard users, or screen-readers. The topic of contrast was also raised but not covered in detail.

- [x] Focus outline styling needed on T3 logo in navbar (2:00)
- [x] Use outline instead of border for focusable elements to avoid layout shift (3:42, again 54:28)
- [x] Remove tabIndex=“-1” from Twitter links to make them focusable (6:40)
- [x] Focus outline styling needed on on vercel badge (10:35)
- [ ] Nested `main` elements semantically incorrect (possibly due to Astro layouts) (15:15)
- [x] Remove alt text on and replace with empty string (alt=“”) on background image SVG to remove from SR (16:24)
- [x] Update aria-label to reflect relevant unique names of tweet cards (22:10)
- [x] Search bar SR simplify aria-label (24:41)
- [x] U Update full clickable cards to include pseudo element with before: to prevent entire content from being read with SR (explained 26:58, tailwind example shown at 47:52)
- [ ] Wrap CLI in a figure element to make it skippable and prevent ASCII art from being read by SR (if this doesn’t work, role=“region” on CLI) (40:52)
- [ ] aria-hidden on library card logos as the names are right next to them, prevents double name reading on SR (46:14)
- [ ] Theme toggle invisible class prevents being focusable (ref: [https://www.scottohara.me/blog/2017/04/14/inclusively-hidden.html ](https://www.scottohara.me/blog/2017/04/14/inclusively-hidden.html)) - possibly should just be buttons (49:20)
- [ ] Remove double link on videos in docs, possibly pseudo element as well (55:20)
- [ ] Focus styling on GitHub contributors (55:45)
- [ ] aria-labels on theme toggle images (56:45)
- [ ] aria-current for docs navbar (57:38)
- [ ] Video links to have better alt text to indicate what the link does rather than what the image is - possibly video embed (58:30)

### Describe alternate solutions

No alternate solution, amazing accessibility feedback that should be implemented. 

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by givensuman**
Partially closes #881

## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

## Changelog

Addressed some points listed in #881 
- aria-current for docs navbar
- aria-labels on theme toggle images
- aria-hidden on library card logos as the names are right next to them, prevents double name reading on SR
- Focus outline styling needed on T3 logo in navbar
---

💯

### CODE REVIEW

**Comment by asrvd** [comment]
would be great if you can work on all the points, we are not in a hurry anyway

**Review by asrvd — APPROVED** [review]

**Comment by asrvd** [comment]
run `pnpm format`

**Comment by juliusmarminge** [comment]
Potential overlap with #886.

Haven't reviewed any of them but judging about your descriptions looks like some stuff is implemented in both

**Comment by asrvd** [comment]
would have been better if one single pr has all the points, maybe they can collab ?

**Comment by juliusmarminge** [comment]
> would have been better if one single pr has all the points, maybe they can collab ?

Sounds scary and guarantees a bunch of conflicts basically 🥶

**Comment by c-ehrlich** [comment]
In general I don't mind having two PRs for one issue that's as broad as accessibility.

But I agree both people should be aware of each other's work to avoid merge conflict nightmares - either work on a branch together, or at least make sure that both aren't "solving" the same bit in different ways.

File: www/src/components/landingPage/stack/stack.astro [hunk 1]
```diff
@@ -24,7 +24,7 @@ import Feature from "./card.astro";
       <div
         class="space-y-10 sm:grid sm:grid-flow-col sm:grid-cols-2 sm:grid-rows-3 sm:gap-x-6 sm:gap-y-14 sm:space-y-0 md:grid-cols-3 md:grid-rows-2 lg:gap-x-8"
       >
-        <Feature title="Next.js" href="https://nextjs.org/">
+        <Feature title="Next.js" href="https://nextjs.org/" aria-hidden={true}>
```
Comment (juliusmarminge) [inline www/src/components/landingPage/stack/stack.astro:27]:
```suggestion
        <Feature
           title="Next.js"
           href="https://nextjs.org/"
           aria-hidden={true}
         >
```

should fix formatting?

Comment (juliusmarminge) [reply www/src/components/landingPage/stack/stack.astro]: guess not :(

**Pull request closed without merging**